api/controllers/SignupController.js

```diff
@@ -72,12 +76,33 @@ module.exports = {
 
         // Result from an update is an array
         if (result.length > 0) {
-          return res.json(result[0]);
+          return result[0];
         } else {
           // Result from a create is the user object
-          return res.json(result);
+          return result;
         }
       })
+      .then(function(result) {
+        // Create custom referral url after a sms user has been created
+        // Use a users first name if available or default to SHINE
+        let uniqueString = result.firstName || 'SHINE';
+        uniqueUrl = ReferralCodes.generateCustomUrl(uniqueString);
+        platformSmsId = result.id;
+        return UserUrl.countByCodeLike(`${uniqueUrl}%`);
```

Comment:
The `countByCodeLike()` method is a function that still needs to be implemented?

Reply:
countByCodeLike is a built in method in sails ORM.